### DISCUSSION TITLE
🚀  GitHub Copilot Workspace: Your Seamless Developer Environment

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#121649] [Copilot News and Announcements]
![Screenshot 2024-04-30 at 3 02 59 PM](https://github.com/community/community/assets/95834130/cd900b47-8594-4f7e-ae93-9d0eae350436)

We’re redefining the developer environment with GitHub Copilot Workspace–where any developer can go from idea, to code, to software in natural language. 

Copilot Workspace represents a radically new way of building software with natural language, and is expressly [designed](https://github.blog/2024-01-17-a-developers-second-brain-reducing-complexity-through-partnership-with-ai/) to deliver–not replace–developer creativity, faster and easier than ever before. With Copilot Workspace we will empower more experienced developers to operate as systems thinkers, and materially lower the barrier of entry for who can build software.

- 🏁  **Seamless Task Initiation**: Open GitHub Copilot Workspace directly from GitHub Issues, Pull Requests, or Repositories, kicking off your development journey effortlessly.
- ⚡  **AI-powered Planning**: From tasks to specifications, Copilot Workspace provides a structured plan tailored to your needs, editable at every step for seamless iteration.
- 💼  **Full Editability and Collaboration**: Modify your action plan and code suggestions as needed, share your workspace with team members via a simple link, and collaborate effortlessly to refine your code.
- 📱  **Integrated Testing and Mobile Compatibility**: Run tests, builds, and checks directly within Copilot Workspace, ensuring robust code, and access the platform from any device, enabling development on-the-go. 😮 Yes! you read that right..
- 🌐  **Empowering Developers Worldwide**: GitHub Copilot Workspace aims to democratize software development, empowering both professional and hobbyist developers to unleash creativity and simplify the development process, driving productivity gains and innovation in the industry.

### [Read the detailed blogpost](https://github.blog/2024-04-29-github-copilot-workspace/) 🔖 

Prefer a short video? 📹  We've got you covered:

https://github.com/community/community/assets/95834130/f4fb1a78-7d5f-4f2f-bbf7-d8e5be6c133e

We've also created tailored FAQ's for your convenience: [FAQ's ](https://githubnext.com/projects/copilot-workspace/) ❓ 

The technical preview for GitHub Copilot Workspace is now live. We can’t wait to see what you will build from here.

**Comment by Seb0098** [comment]
![1000000074](https://github.com/community/community/assets/156356321/08d641f3-886b-4679-8b7f-9e49f2377451)

**Comment by juanlittledevil** [comment]
I have been using copilot for about 2 months now. and the experience has been less than pleasant. For starters, unlike chatGPT 3.5 it doesn't retain context from conversations. This is frustrating because I end up having to repeat myself over and over and over again. Then, it more often than not suggests solutions wildly different than what is implemented. Another thing that is also very annoying is that we can't setup default configuration for how the AI should interact, respond, etc like that ChatGPT. Overall I've gotten better results from pasting code into ChatGPT than using Copilot. I'm this close to canceling my subscription and asking for my money back. So far this is NOT useful!

**Reply by Hatseflatsh** [reply]
I am actually doing the same. I just copy paste from ChatGPT :D

**Reply by Kellz99** [reply]
.

**Reply by JeanLeeJeffrson** [reply]
Chce to mít chatgpt rovnou v aplikaci webu programu s nápovědou a proxy server s otevřenou možnou podporu rovnou na komunitu nebo evropské orgány pro bezpečnost

**Reply by dwahlqvist** [reply]
I agree. Whenever I want to do something more than basic, I've usually wasted more time explaining the issue and correcting the AI, compared to doing it myself + google + stack overflow. What's most frustrating is that after 3-4 times back and forth, it often starts suggesting the same incorrect answers I already discarded.
It often suggests code that doesn't run or make sense. 
It doesn't have all relevant context or even ability to scan my project, even when using "@workspace" (by the way, is that the same "workspace" as talked about above or are they repurposing the term?) 

It's not worthless but more like Intellisense++.

Looking at it from the bright side, I don't think we'll be replaced by AI anytime soon ;)

**Reply by goldieowner17** [reply]
Hi @JeanLeeJeffrson, thanks for participating!

We take our [[Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md)] very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be hiding any comments in languages other than English for now, including this one.

**Reply by JeanLeeJeffrson** [reply]
> Hi @JeanLeeJeffrson, thanks for participating!
> 
> We take our [[Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md)] very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be hiding any comments in languages other than English for now, including this one.

It wants to have a chatgpt straight in the web application program with help and a proxy server with open possible support straight to the community or European security authorities

**Comment by Olvinda** [comment]
### 

> 

> [](url)

**Comment by Kellz99** [comment]
> I have been using copilot for about 2 months now. and the experience has been less than pleasant. For starters, unlike chatGPT 3.5 it doesn't retain context from conversations. This is frustrating because I end up having to repeat myself over and over and over again. Then, it more often than not suggests solutions wildly different than what is implemented. Another thing that is also very annoying is that we can't setup default configuration for how the AI should interact, respond, etc like that ChatGPT. Overall I've gotten better results from pasting code into ChatGPT than using Copilot. I'm this close to canceling my subscription and asking for my money back. So far this is NOT useful!

**Comment by Mehraaan14** [comment]
> ![Screenshot 2024-04-30 at 3 02 59 PM](https://github.com/community/community/assets/95834130/cd900b47-8594-4f7e-ae93-9d0eae350436)
> 
> We’re redefining the developer environment with GitHub Copilot Workspace–where any developer can go from idea, to code, to software in natural language. 
> 
> 
> Copilot Workspace represents a radically new way of building software with natural language, and is expressly [designed](https://github.blog/2024-01-17-a-developers-second-brain-reducing-complexity-through-partnership-with-ai/) to deliver–not replace–developer creativity, faster and easier than ever before. With Copilot Workspace we will empower more experienced developers to operate as systems thinkers, and materially lower the barrier of entry for who can build software.
> 
> - 🏁  **Seamless Task Initiation**: Open GitHub Copilot Workspace directly from GitHub Issues, Pull Requests, or Repositories, kicking off your development journey effortlessly.
> - ⚡  **AI-powered Planning**: From tasks to specifications, Copilot Workspace provides a structured plan tailored to your needs, editable at every step for seamless iteration.
> - 💼  **Full Editability and Collaboration**: Modify your action plan and code suggestions as needed, share your workspace with team members via a simple link, and collaborate effortlessly to refine your code.
> - 📱  **Integrated Testing and Mobile Compatibility**: Run tests, builds, and checks directly within Copilot Workspace, ensuring robust code, and access the platform from any device, enabling development on-the-go. 😮 Yes! you read that right..
> - 🌐  **Empowering Developers Worldwide**: GitHub Copilot Workspace aims to democratize software development, empowering both professional and hobbyist developers to unleash creativity and simplify the development process, driving productivity gains and innovation in the industry.
> 
> ### [Read the detailed blogpost](https://github.blog/2024-04-29-github-copilot-workspace/) 🔖 
> 
> Prefer a short video? 📹  We've got you covered:
> 
> 
> https://github.com/community/community/assets/95834130/f4fb1a78-7d5f-4f2f-bbf7-d8e5be6c133e
> 
> We've also created tailored FAQ's for your convenience: [FAQ's ](https://githubnext.com/projects/copilot-workspace/) ❓ 
> 
> The technical preview for GitHub Copilot Workspace is now live. We can’t wait to see what you will build from here.
> 
> ### [Sign up now](https://githubnext.com/projects/copilot-workspace) 🚀 
>

**Comment by Mehraaan14** [comment]
> ![1000000074](https://github.com/community/community/assets/156356321/08d641f3-886b-4679-8b7f-9e49f2377451)
>

**Comment by CharlonTank** [comment]


**Reply by JeanLeeJeffrson** [reply]
> When I try to open copilot workspace on an issue I get:
> 
> API rate limit exceeded for user ID 4601603. If you reach out to GitHub Support for help, please include the request ID 0540:16F13C:24776E:420A59:663C933A and timestamp 2024-05-09 09:11:24 UTC.
> 
> Since it doesn't give a support link, I put that here hopefully to get an answer :)
> 
> Thank you!

Test with security Cyber. This Is problem in Speed network?
